module/plugins/captcha/ReCaptcha.py

```diff
@@ -354,3 +356,45 @@ def _challenge_v2(self, key, secure_token=None):
             self.fail(_("Recaptcha max retries exceeded"))
 
         return result, challenge
+
+    # solve interactive captcha (javascript required), use when non-JS captcha fallback for v2 is not allowed
+    def _challenge_v2_js(self, key):
+        self.log_debug("challenge ReCaptcha v2 interactive")
+        captchaManager = self.pyload.captchaManager
+
+        try:
+            time_ref = ("%.2f" % time.time())[-6:].replace(".", "")
+            filename = "captcha_request_%s_%s.nofile" % (self.pyfile.plugin.__name__, time_ref)
+            # we don't have an actual image to decrypt -> pass the URL and the site-key for further use in plugins/hooks
+            img = {'url': self.pyfile.url, 'sitekey': self.retrieve_key(None)}
```

Comment:
why not use `key` as it has been passed to the method. or remove key from the parameters.